bindings/yarp.i

```diff
@@ -1535,6 +1547,47 @@ typedef yarp::os::BufferedPort<ImageRgbFloat> BufferedPortImageRgbFloat;
 #endif
 }
 
+%extend yarp::sig::VectorOf<int> {
+
+    // This in not a real constructor actually, it is converted by swig to a function returning a pointer.
+    // See: http://www.swig.org/Doc3.0/CPlusPlus11.html#CPlusPlus11_initializer_lists
+    VectorOf<int>(const std::vector<int>& values)
+    {
+        VectorOf<int>* newVec = new VectorOf<int>(0);
+        newVec->reserve(values.size());
+        for (const auto& element : values) {
+            newVec->push_back(element);
+        }
+        return newVec;
+    }
+
+    int get(int j)
+    {
+        return self->operator [](j);
+    }
+
+    void set(int j, int v)
+    {
+        self->operator [](j) = v;
+    }
+
+
+
+#ifdef SWIGPYTHON
+    void __setitem__(int key, int value) {
+        self->operator[](key) = value;
+    }
+
+    int __getitem__(int key) {
+        return self->operator[](key);
+    }
+
+    int __len__() {
+        return self->length();
+    }
+#endif
+}
+
```

Comment:
I wonder if this could become something like
```c++
%extend
template <typename T>
yarp::sig::VectorOf<T> {
// ...
```
It seems quite inconvenient to have to duplicate the code for each type that we want to support

Reply:
Tried it but seems not working (it fires up a syntax error).